src/execution_plan/ops/shared/create_functions.c

```diff
@@ -377,22 +377,22 @@ void PendingCreationsFree
 		pending->created_edges = NULL;
 	}
 
-	if(pending->node_attributes) {
+	if(pending->node_attributes != &NULL_ATTRIBUTE_SET && pending->edge_attributes) {
 		uint prop_count = array_len(pending->node_attributes);
 		for(uint i = 0; i < prop_count; i ++) {
 			AttributeSet_Free(pending->node_attributes + i);
 		}
 		array_free(pending->node_attributes);
-		pending->node_attributes = NULL;
+		pending->node_attributes = &NULL_ATTRIBUTE_SET;
 	}
 
-	if(pending->edge_attributes) {
+	if(pending->edge_attributes != &NULL_ATTRIBUTE_SET && pending->edge_attributes) {
 		uint prop_count = array_len(pending->edge_attributes);
  		for(uint i = 0; i < prop_count; i ++) {
  			AttributeSet_Free(pending->edge_attributes + i);
  		}
 		array_free(pending->edge_attributes);
-		pending->edge_attributes = NULL;
+		pending->edge_attributes = &NULL_ATTRIBUTE_SET;
 	}
 }
```

Comment:
I think no changes are required here, can we revert back ?

Reply:
Ok. Changes reverted.